### ISSUE TITLE
BROWSER switch not working

### ISSUE THREAD

**Issue by dweidenfeld** [pbock/fahrplan#7]
Hey,

if your packaging the app within a webpack project, the BROWSER switch for your request library is not working.

The best way to check if your in a browser is

``` javascript
'use strict';

if (typeof window !== 'undefined') module.exports = require('./browser');
else module.exports = require('./node');
```

**Comment by derhuerst** [comment]
> The best way to check if your in a browser is […].

There is no need to check for a browser environment. Feature sniffing is considered a better practice. [isomorphic-fetch](https://github.com/matthew-andrews/isomorphic-fetch#isomorphic-fetch-) implements the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) cross-platform.

**Comment by dweidenfeld** [comment]
Yes I agree but it this is a bit smaller change ;) ... but I'm fine with both solutions.

**Comment by pbock** [comment]
Thanks for letting me know, embarrassingly I don't think I tested what happens when you package the client with Webpack.

I agree with @derhuerst in principle: isomorphic-fetch is lovely, and I've been using it in my standalone apps recently. My rationale against using it in a project that others depend on is that I want to avoid external dependencies if at all possible.

This is partly to keep the file size/dependency tree small (and because I only need a small part of what `fetch()` offers), but mainly because I'm not sure how the licensing would work: I'm distributing the client using [Universal Module Definition](https://github.com/umdjs/umd), so in addition to being able to `require()`/`include` it, you can also include it via `<script src="fahrplan.js"></script>`. This means that any code I depend on would get bundled into `fahrplan.js`, and while that's fine under the terms of the MIT License, I'm _really_ not sure how/where to credit the authors.

Of course, I could just ignore the dependency, at least for the bundled file, and tell people to ensure that there's a global `fetch()` (like I do with `Promise`), but that feels wrong.

So for now, I think I'll take @dweidenfeld's approach and check for the environment. I didn't do that previously because I didn't understand how to prevent Webpack from mocking node.js's `http`/`https` modules.

**Comment by pbock** [comment]
Fixed in v0.3.4/commit 0290067.
